python/ray/tests/kuberay/test_autoscaling_e2e.py

```diff
@@ -138,6 +122,30 @@ def _get_ray_cr_config(
         gpu_group["groupName"] = "fake-gpu-group"
         config["spec"]["workerGroupSpecs"].append(gpu_group)
 
+        # Substitute images.
+        for group_spec in config["spec"]["workerGroupSpecs"] + [
+            config["spec"]["headGroupSpec"]
+        ]:
+            containers = group_spec["template"]["spec"]["containers"]
+
+            # Will assume the example config maintains this invariant:
```

Comment:
uh, this is dangerous -- you should add an `assert ray_container["name"] == "ray-head"` or similar here

Reply:
Will do -- will also rename the worker containers "ray-worker" from what is now (I think it's "ml" or something)

Reply:
Ray container for workers renamed "ray-worker". Test now asserts that the first container is one of `["ray-worker", "ray-head"]`.